rlclientlib/logger/eventhub_client.h

```diff
@@ -91,18 +78,11 @@ namespace reinforcement_learning {
     eventhub_client& operator=(eventhub_client&&) = delete;
 
   private:
-    web::http::client::http_client _client;
-
+    std::unique_ptr<i_http_client> _client;
```

Comment:
Could you possibly add a clarifying comment for where we pass this object in that the class takes ownership of it and will delete at the end of lifetime. We have a lot of instances where we do not manage lifetime so we should be explicit here